include/tvm/relay/vm/vm.h

```diff
@@ -0,0 +1,299 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file tvm/relay/runtime/runtime.h
+ * \brief Abstract device memory management API
+ */
+#ifndef TVM_RELAY_RUNTIME_H_
+#define TVM_RELAY_RUNTIME_H_
+
+#include <vector>
+#include <memory>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/logging.h>
+#include <tvm/runtime/memory_manager.h>
+
```

Comment:
move data structure to tvm/runtime/vm, let us keep only the high level data structure in header, move most of the implementations(Frame, Stack etc) to the src